src/features/pages/home/live-pricing/table-component/live-market-table.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import {
+    flexRender,
+    getCoreRowModel,
+    getSortedRowModel,
+    useReactTable,
+} from '@tanstack/react-table'
+import type { SortingState } from '@tanstack/react-table'
+import { TAvailableLiveMarkets, TMarketData } from '../types'
+import useLiveColumns from '../use-live-columns'
+import { table_row_header, table_row_data } from './live-pricing.module.scss'
+import Flex from 'features/components/atoms/flex-box'
+import useRegion from 'components/hooks/use-region'
+import Link from 'features/components/atoms/link'
+import { Localize } from 'components/localization'
+
+export type TLiveMarketTableProps = {
+    selected_market: TAvailableLiveMarkets
+    link_to: string
+}
+
+const LiveMarketTable = ({ selected_market, link_to }: TLiveMarketTableProps) => {
+    const { is_eu } = useRegion()
+    const [initial_loaded, setInitialLoaded] = useState(false)
+
+    const [rawMarketsData, setRawMarketsData] = useState()
+    const TABLE_VISIBLE_ROWS = 5
+    const [markets_data, setMarketsData] = useState(() => {
+        const temp = new Map<TAvailableLiveMarkets, TMarketData[]>()
+        return temp
+    })
+    const intervalRef = useRef(null)
+
+    const [sorting, setSorting] = React.useState<SortingState>([])
+
+    const region = useMemo(() => {
+        return is_eu
+            ? 'https://deriv-static-pricingfeed.firebaseio.com/eu.json'
+            : 'https://deriv-static-pricingfeed.firebaseio.com/row.json'
+    }, [is_eu])
+
+    const getData = useCallback(async () => {
+        const rawResponse = await fetch(region)
+        const response = await rawResponse.json()
+        setRawMarketsData(response)
+    }, [region])
+
+    useEffect(() => {
+        if (!initial_loaded) {
+            getData()
+            setInitialLoaded(true)
+        }
+        intervalRef.current = setInterval(getData, 10000)
+        return () => {
+            clearInterval(intervalRef.current)
+        }
+    }, [getData, initial_loaded])
+
+    const updateData = useCallback(() => {
+        if (rawMarketsData) {
+            const stocks = rawMarketsData.market['stocks']
+            const indices = rawMarketsData.market['indices']
+            const stocks_indices = Object.assign(stocks, indices)
+            const res = { ...rawMarketsData.market, indices: { ...stocks_indices } }
+
+            Object.keys(res).map((item) => {
+                if (item == selected_market) {
+                    const selected_market_data = res[item]
+                    const result = Object.values(selected_market_data)
+                    setMarketsData(result)
+                }
+            })
+        }
+    }, [rawMarketsData, selected_market])
```

Comment:
@NikitK-deriv This approach was suggested by @mohsen-deriv. I tried refactoring, but it breaks when switching from tabs. Need refactoring this file, since it's an urgent task let's keep as it is and need your help on this so we can do as an improvement.